lib/bracket-matcher.coffee

```diff
@@ -218,3 +224,6 @@ class BracketMatcher
 
   unsubscribe: ->
     @subscriptions.dispose()
+
+  getScopedSetting: (key) ->
+    atom.config.get(key, scope: @editor.getLastCursor().getScopeDescriptor())
```

Comment:
> If performance is a problem, I would change it to getRootDescriptor().

Yeah, I believe we shouldn't regress in terms of performance, and `2s` is a pretty serious slowdown. I'd still say to use `getLastCursor()`, as the behavior would be more accurate in the case of a single selection.

Can we write a spec for the wrap selection behavior as well as backspacing when the `autocompleteBrackets` configuration is disabled in scope? :white_check_mark: